packages/bitcore-node/src/modules/ripple/api/csp.ts

```diff
@@ -112,22 +112,32 @@ export class RippleStateProvider extends InternalStateProvider implements IChain
     const { network, time } = params;
     const date = new Date(time || Date.now()).toISOString();
     const ledger = await new Promise((resolve, reject) => {
-      const url = this.config[network].provider.dataHost + '/v2/ledgers/' + date;
-      request.get({ url, json: true }, (err, _, body) => {
-        if (err) {
-          return reject(err);
-        } else {
-          return resolve({
-            ...body.ledger,
-            chain: this.chain,
-            network,
-            hash: body.ledger.ledger_hash,
-            height: body.ledger.ledger_index,
-            previousBlockHash: body.ledger.parent_hash,
-            timeNormalized: new Date(body.ledger.close_time * 1000)
-          });
-        }
-      });
+      try {
+        const url = this.config[network].provider.dataHost + '/v2/ledgers/' + date;
+        request.get({ url, json: true }, (err, _, body) => {
+          try {
+            if (err) {
+              return reject(err);
+            } else if (body == null) {
```

Comment:
Is it fully null? May want to do `if(!body.ledger)`. And would it make sense to 404 in that case? Is there an example of what is coming back when there is no ledger property? There may be a constructive error from ripple.